controllers/userController.js

```diff
@@ -101,6 +102,41 @@ router.post('/:id/password', verifyToken, (req, res) => {
 	});
 });
 
+// Changes a don's information (only accessible by dons)
+router.post('/:id/doninfo', verifyToken, (req, res) => {
+	if(req.user.role&1 != 1) {
+		console.error("User "+ req.user._id +"is requesting don info when user is not a don");
+		//requestor is not a don
+		res.status(403);
+	}
+	User.findOne({_id: req.params.id}).then(user => {
+		if(user.role&1 != 1){
+			//user is not a don
+			res.status(403);
+		}else{
+			user.donInfo = req.body
+			user.update(user);
+			//set timer to turn off don automagically
+			if(user.donInfo.isOn){
+				setTimer(user.donInfo.clockOut, ''+user._id, {},() =>{
```

Comment:
That's fairly long for an anonymous function. Maybe throw a method on the user model, which is just passed into `setTimer` here?

Also is `''+user._id` trying to turn something into a string? If so you should just be able to call `user._id.toString()` instead.

Reply:
This is me just not knowing javascript, I'll update this